extensions/wikia/ImageReview/ImageListGetter.class.php

```diff
@@ -0,0 +1,368 @@
+				$dbh->freeResult( $results );
+			}
+
+			return array_shift( $rows );
+		};
+	}
+
+	private function assignImageToUser( $reviewStart, $state, stdClass $imageRecord ) {
+		// Determine what our next state should be
+		$targetState = ImageReviewStates::IN_REVIEW;
+
+		if ( $state == ImageReviewStates::QUESTIONABLE ) {
+			$targetState = ImageReviewStates::QUESTIONABLE_IN_REVIEW;
+		} elseif ( $state == ImageReviewStates::REJECTED ) {
+			$targetState = ImageReviewStates::REJECTED_IN_REVIEW;
+		}
+
+		$dbw = $this->getDatawareDB( DB_MASTER );
+		$query = ( new WikiaSQL() )
+			->UPDATE( 'image_review' )
+				->SET( 'reviewer_id', $this->userId )
+				->SET( 'review_start', $reviewStart )
+				->SET( 'state', $targetState )
+			->WHERE( 'reviewer_id' )->IS_NULL()
+				->AND_( 'wiki_id' )->EQUAL_TO( $imageRecord->wiki_id )
+				->AND_( 'page_id' )->EQUAL_TO( $imageRecord->page_id );
+
+		if ( $state == ImageReviewStates::QUESTIONABLE ||
+		     $state == ImageReviewStates::REJECTED ) {
+			$query->SET( 'review_end', '0000-00-00 00:00:00' );
+		}
+
+		$query->run( $dbw );
+		$affectedRows = $dbw->affectedRows();
+		$dbw->commit();
+
+		return $affectedRows;
+	}
+
+	/**
+	 * Given data from the image_review table, pull additional details like image URL, extension
+	 * and wiki URL.
+	 *
+	 * @param stdClass $imageRecord A row from the image_review table
+	 *
+	 * @return array
+	 */
+	private function getImageData( stdClass $imageRecord ) {
+		$imageInfo = $this->getBaseImageInfo( $imageRecord->wiki_id, $imageRecord->page_id );
+		if ( !empty( $imageInfo['error'] ) ) {
+			return $imageInfo;
+		}
+
+		$wikiRow = WikiFactory::getWikiByID( $imageRecord->wiki_id );
+		$cityUrl = !empty( $wikiRow ) && $wikiRow->city_url ? $wikiRow->city_url : '';
+
+		return [
+			'wikiId' => $imageRecord->wiki_id,
+			'pageId' => $imageRecord->page_id,
+			'state' => $imageRecord->state,
+			'src' => $imageInfo['src'],
+			'url' => $imageInfo['page'],
+			'extension' => $imageInfo['extension'],
+			'priority' => $imageRecord->priority,
+			'flags' => $imageRecord->flags,
+			'isthumb' => true,
+			'wiki_url' => $cityUrl,
+		];
+	}
+
+	private function getBaseImageInfo( $wikiId, $pageId ) {
+		$result = $this->getImageGlobalFile( $wikiId, $pageId );
+		if ( !empty( $result['error'] ) ) {
+			return $result;
+		}
+		/** @var GlobalFile $imageGlobalFile */
+		$imageGlobalFile = $result['file'];
+		/** @var GlobalTitle $imageGlobalPage */
+		$imageGlobalPage = $result['page'];
+
+		$thumbUrl = $imageGlobalFile->getUrlGenerator()
+			->width( self::IMAGE_REVIEW_THUMBNAIL_SIZE )
+			->height( self::IMAGE_REVIEW_THUMBNAIL_SIZE )
+			->thumbnailDown()
+			->url();
+
+		return [
+			'src' => $thumbUrl,
+			'page' => $imageGlobalPage->getFullURL(),
+			'extension' => $imageGlobalFile->getMimeType(),
+		];
+	}
+
+	/**
+	 * @param $pageId
+	 * @param $wikiId
+	 *
+	 * @return array
+	 *
+	 * @throws Exception
+	 */
+	private function getImageGlobalFile( $wikiId, $pageId ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)